src/manifest.rs

```diff
@@ -183,6 +164,75 @@ impl Manifest {
         data.parse()
     }
 
+    /// Get the specified table from the manifest.
+    pub fn get_table<'a>(
+        &'a mut self,
+        table_path: &[String],
+    ) -> Result<&'a mut BTreeMap<String, toml::Value>, ManifestError> {
+
+        /// Descend into a manifest until the required table is found.
+        fn descend<'a>(
+            input: &'a mut BTreeMap<String, toml::Value>,
+            mut path: VecDeque<&String>,
```

Comment:
Can we avoid allocation?
I think, if we make `path: &[String]`, then 
`if let Some(segment) = path.pop_front() {` can be replaced with `if !path.is_empty() { let segment = path[0];`
and `descend(value, path)` with `descend(value, &path[1..])`